api/roleTickets/roleTicketsRouter.js

```diff
@@ -16,4 +116,279 @@ router.get('/', authRequired, adminRequired, (req, res, next) => {
     });
 });
 
+/**
+ * @swagger
+ * /role-tickets/{role_ticket_id}:
+ *  get:
+ *    summary: Get details about a single role ticket
+ *    tags:
+ *      - role ticket
+ *    security:
+ *      - okta: []
+ *    parameters:
+ *      - in: path
+ *        name: role_ticket_id
+ *        schema:
+ *          type: integer
+ *        required: true
+ *        description: Numeric ID of the role ticket to look for
+ *    responses:
+ *      '200':
+ *        description: Information about a specific role ticket
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Role Ticket'
+ *      '401':
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      '404':
+ *        description: Role Ticket with the given ID could not be found
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: Error message returned by the API
+ *                  example: 'Role Ticket with ID 1 not found!'
+ */
+
+router.get(
+  '/:role_ticket_id',
+  authRequired,
+  adminRequired,
+  checkRoleTicketIdExists,
+  (req, res, next) => {
+    const id = req.params.role_ticket_id;
+    RoleTickets.findByRoleTicketById(id)
```

Comment:
Seems odd to call the database twice for the same data -- your `checkRoleTicketIdExists()` middleware stores the fetched result in `req.roleTicket`, so all you need to do is include it in a response.

Reply:
Solid catch, I've made modifications to use the fetched from the middleware.